pkg/k8s/service_cache.go

```diff
@@ -466,12 +475,13 @@ func (s *ServiceCache) UpdateEndpoints(newEndpoints *Endpoints, swg *lock.Stoppa
 	if ok && serviceReady {
 		swg.Add()
 		s.emitEvent(ServiceEvent{
-			Action:       UpdateService,
-			ID:           esID.ServiceID,
-			Service:      svc,
-			Endpoints:    endpoints,
-			OldEndpoints: oldEPs,
-			SWG:          swg,
+			Action:                        UpdateService,
+			ID:                            esID.ServiceID,
+			Service:                       svc,
+			Endpoints:                     endpoints,
+			EndpointLastChangeTriggerTime: triggerTime,
```

Comment:
Based on the comment next to the metric, I am not sure if this is the correct place to measure the time from.
```
	// KubernetesNetworkProgrammingLatency is defined as the time it took to program the network - from the time
	// the service or pod has changed to the time the change was propagated and the proper kube-proxy/cilium
	// rules were synced.
```

There is bound to be some time between K8s object change and the time this event gets propagated to Cilium.

I think the correct way to get this data is to implement using `ManagedFields` in slim Service objects that we use which is currently unimplemented: https://github.com/cilium/cilium/blob/main/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1300 and use `Time` field from these to figure out when object was last changed.

Reply:
@weiwei02 can you tell me how you resolved this?